peachjam/forms.py

```diff
@@ -181,6 +181,11 @@ def _save_m2m(self):
             if self.instance.document.extract_content_from_source_file():
                 self.instance.document.save()
 
+                # if the file is changed, we need delete the existing pdf and re-generate
+                self.instance.file_as_pdf = None
```

Comment:
perhaps `self.instance.file_as_pdf.delete()` which also deletes the file from storage (and saves the model by default).

Reply:
Changed. Haven't used `file.delete` before.